### PR TITLE
fix: update legends test for iox

### PR DESCRIPTION
**Pull request by wdoconnell**
Part of #6562 

This PR:
1. Updates the legends tests to work with IOx.
2. Unskips tests that weren't being run due to Cypress firefox compatibility issues (we only test Chrome now).
3. Fixes a bug where a new iox user with access to old data explorer but none of the other features could still see the 'Save As' button, even if there was nowhere to save the query.

### Checklist

Authors and Reviewer(s), please verify the following:

- [X] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [X] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [X] Documentation updated or issue created (provide link to issue/PR)
- [X] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [X] Feature flagged, if applicable - NO

### CODE REVIEW

File: cypress/e2e/shared/legends.test.ts [hunk 1]
```diff
@@ -540,7 +549,9 @@ describe.skip('Legends', () => {
           cy.getByTestID('page-header').click().type(cellName)
 
           // Save it to a dashboard
+          cy.get('.veo-contents').click() // click out of inline editor
           cy.getByTestID('save-cell--button').click()
```
Comment (wdoconnell) [inline cypress/e2e/shared/legends.test.ts:553]: This is needed because the save-cell button isn't properly selected by Cypress unless you click out of the inline editor. This is the same fix we've employed on other tests here.

File: src/dataExplorer/components/DataExplorerPage.tsx [hunk 2]
```diff
@@ -137,7 +137,7 @@ const DataExplorerPage: FC = () => {
     (scriptQueryBuilder && isFlagEnabled('newDataExplorer')) || isNewIOxOrg
 
   const shouldShowSaveAsButton =
-    !isNewIOxOrg ||
+    !useSelector(selectIsNewIOxOrg) ||
     shouldShowNotebooks ||
```
Comment (wdoconnell) [inline src/dataExplorer/components/DataExplorerPage.tsx:141]: This edit is a minor bug fix that requires some explanation because it's pretty far down the logic tree of what can happen within the matrix of feature overrides.

`shouldShowSaveAsButton` controls whether the "Save As" button on data explorer displays. This variable evaluates to `true` and permits the display of the "Save As" button in data explorer if the user is _not_ in an IOx org; or, if the user _is_ in an IOx org, but still has access to any one of the following: notebooks, tasks, dashboards, or variables. That can happen if the user is in the IOx lighthouse program, or has a feature override flag enabled. 

The logic wasn't right here for a specific edge case that appeared during the fixed e2e tests:
1. You're in an IOx org and
2. `showOldDataExplorerInNewIOx` is on and
3. All other overrides are off

In that case, `!isNewIOxOrg` incorrectly evaluates to `true`, since the override flag for data explorer is _not_ off (`!isFlagEnabled('showOldDataExplorerInNewIOx')`). This shows the "Save As" button even though the resulting overlay wouldn't offer you any place to save the query. The "Save As" button shouldn't care whether you have the `showOldDataExplorerInNewIOx` flag on or off. The "Save As" button should only display if you're in a TSM/IOx lighthouse org; or one of the tasks/notebooks/dashboards/variables overrides is on.

File: src/flows/selectors/flowsSelectors.ts [hunk 3]
```diff
@@ -11,6 +11,10 @@ export const selectNotebooks = (state: AppState): Notebook[] => {
 }
 
 export const selectShouldShowNotebooks = (state: AppState): boolean => {
+  if (isFlagEnabled('showNotebooksForCI')) {
+    return true
+  }
+
```
Comment (wdoconnell) [inline src/flows/selectors/flowsSelectors.ts:17]: TLDR version: If `showNotebooksForCI` is on, this selector should always return true.

--
Less TLDR version: 

This selector was producing issues in this e2e test because, when trying to force the display of notebooks for CI testing in Cloud, we would get this result:
1. `if (!CLOUD)` is `false` - skipped.
2. `if (!wasCreatedBeforeIOxCutoff & !isFlagEnabled('showNotebooksForCI'))` is false (the flag is on) - skipped.
3. Selector returns whether the number of notebooks in state is `> 0`.

3. is always false if the test hasn't set up a notebook yet. So we need to move the logic that forces the display of notebooks based on the `showNotebooksForCI` flag up, so that Cypress can navigate to the notebooks page to create whatever notebook it needs for the test.

**Review by appletreeisyellow — APPROVED** [review]
Looks good to me! The comments help a lot with why the changes were made!

File: src/dataExplorer/components/DataExplorerPage.tsx (hunk 2 above)
Comment (wiedld) [reply src/dataExplorer/components/DataExplorerPage.tsx:141]: We are getting some weird behaviors, because we have conditionals (including flags and iox org checking) at multiple levels in our code. In the scripts tests, we ran into a conflict between the `isFlagEnabled('showOldDataExplorerInNewIOx')` and the AppSettings global state of `scriptQueryBuilder`.

We are merging in a new fixture with [this PR ](https://github.com/influxdata/ui/pull/6593#discussion_r1099621263)(hopefully merged today), which will have a new `cy.mockIsCloud2Org()` which returns an older createdAt date for the org.

This means that any deprecated features based on org createdAt date, will be turned back on if we use this in the beforeEach(). This includes features deprecated on iox (e.g. marty release), and also in the future could be used on tsm if we deprecate features prior to the migration to iox backend for those folks.

Comment (wiedld) [reply src/dataExplorer/components/DataExplorerPage.tsx:141]: People now have the choice of using the flags (e.g. `showDashboardsInNewIOx`) as needed for some tests. But for features that have nested conditionals throughout shared logic (a.k.a. old and new data explorer), using this `cy.mockIsCloud2Org()`, followed by `cy.switchToDataExplorer('old')`may be easier.

File: src/flows/selectors/flowsSelectors.ts (hunk 3 above)
Comment (wiedld) [reply src/flows/selectors/flowsSelectors.ts:18]: Or we could mock a notebook creation. See the `release.test`. 
https://github.com/influxdata/ui/blob/master/cypress/e2e/cloud/releases.test.ts#L165-L171

Comment (wdoconnell) [reply src/flows/selectors/flowsSelectors.ts:18]: 👍

**Comment by wdoconnell** [comment]
Removed the proposed changes to the notebooks selector after discussion with @wiedld. We actually don't need them for this specific PR, as it turns out.

**Review by appletreeisyellow — APPROVED** [review]

**Comment by wdoconnell** [comment]
Need a quick re-review to test a fix - in merge queue, not working with OSS, likely due to the cloud-only FFs being switched on.

**Review by appletreeisyellow — DISMISSED** [review]

**Review by blegesse-git — APPROVED** [review]

**Comment by wdoconnell** [comment]
Rebasing this because tests have been updated (tasks has been split into two tests) since the creation of this PR.

**Comment by wdoconnell** [comment]
This one required further edits because it was failing in merge queue, and the test was so long that was leaving no artifact.

I will merge this when I'm back online on Thursday.

**Review by appletreeisyellow — DISMISSED** [review]

**Review by appletreeisyellow — APPROVED** [review]
